Makefile

```diff
@@ -540,7 +537,6 @@ EXAMPLE_PROVIDER_IMG ?= $(REGISTRY)/example-provider-controller
 .PHONY: docker-build-example-provider
 docker-build-example-provider: ## Build the docker image for example provider
 	DOCKER_BUILDKIT=1 docker build --pull --build-arg goproxy=$(GOPROXY) --build-arg ARCH=$(ARCH) . -f ./cmd/example-provider/Dockerfile -t $(EXAMPLE_PROVIDER_IMG)-$(ARCH):$(TAG)
-	sed -i'' -e 's@image: .*@image: '"${EXAMPLE_PROVIDER_IMG}-$(ARCH):$(TAG)"'@' ./config/ci/manager/manager_image_patch.yaml
```

Comment:
IIRC we needed this to build the docker container for some old e2e, if those have been removed we can probably safely delete the example provider

Reply:
I will open a good first issue for this

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/3991